exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -316,9 +366,18 @@ public EmulatedNode getEmulatedNode() {
 
   @Override
   protected void disposeInternal() {
+    closeSnapshotCleaners();
     nativeFreeTestKit(nativeHandle.get());
   }
 
+  private void closeSnapshotCleaners() {
+    try {
+      snapshotCleaner.close();
+    } catch (CloseFailuresException e) {
+      throw new RuntimeException(e);
```

Comment:
(1) Inappropriate exception type: IllegalStateException would be more suitable;
(2) The testkit itself will be leaked if such exception occurs. We might add it as a CleanAction
to ensure reliable destruction, but I think using cleaner here (in explicitly closeable classes
that control that Cleaner) would add unnecessary indirection, and try/finally will work better:

```java
 protected void disposeInternal() {
    try {
      snapshotCleaner.close();
    } catch (CloseFailuresException e) {
      throw …
    } finally {
      nativeFreeTestKit(nativeHandle.get());
    }
```